From 934e5aacab3a99cf205d5c3e67deeaf8b29875af Mon Sep 17 00:00:00 2001
From: Guillaume Desmottes <guillaume.desmottes@collabora.co.uk>
Date: Thu, 31 May 2012 09:28:41 +0000
Subject: notificationDaemon.js: convert the hints dict at the beginning of the function

It's used right away to discard some Empathy notifications.

This regression has been introduced during the 3.4 cycle when 'hints' has been
turned to a GVariant.

https://bugzilla.gnome.org/show_bug.cgi?id=675370
---
diff --git a/js/ui/notificationDaemon.js b/js/ui/notificationDaemon.js
index 96f1ec9..cf89733 100644
--- a/js/ui/notificationDaemon.js
+++ b/js/ui/notificationDaemon.js
@@ -221,6 +221,13 @@ const NotificationDaemon = new Lang.Class({
         let [appName, replacesId, icon, summary, body, actions, hints, timeout] = params;
         let id;
 
+        for (let hint in hints) {
+            // unpack the variants
+            hints[hint] = hints[hint].deep_unpack();
+        }
+
+        hints = Params.parse(hints, { urgency: Urgency.NORMAL }, true);
+
         // Filter out chat, presence, calls and invitation notifications from
         // Empathy, since we handle that information from telepathyClient.js
         if (appName == 'Empathy' && (hints['category'] == 'im.received' ||
@@ -249,13 +256,6 @@ const NotificationDaemon = new Lang.Class({
             }
         }
 
-        for (let hint in hints) {
-            // unpack the variants
-            hints[hint] = hints[hint].deep_unpack();
-        }
-
-        hints = Params.parse(hints, { urgency: Urgency.NORMAL }, true);
-
         // Be compatible with the various hints for image data and image path
         // 'image-data' and 'image-path' are the latest name of these hints, introduced in 1.2
 
--
cgit v0.9.0.2
